rust/lance/src/dataset/transaction.rs

```diff
@@ -1759,6 +1773,7 @@ impl Transaction {
                 .fields
                 .iter()
                 .all(|field_id| field_ids.contains(field_id))
+                || existing_index.name == FRAG_REUSE_INDEX_NAME
```

Comment:
Does this mean the frag reuse indices will never go away? How long will we retain them?